test/cpp/jit/test_gpu.cpp

```diff
@@ -671,8 +671,8 @@ void testGPU_FusionCodeGen() {
   TensorView* tv0 = makeDummyTensor(3);
 
   new BinaryOp(BinaryOpType::Add, tv0, new Float(0.0), new Float(1.0));
-  TensorView* tv1 = static_cast<TensorView*>(add(tv0, new Float(2.0)));
-  TensorView* tv2 = static_cast<TensorView*>(add(tv1, new Float(3.0)));
+  TensorView* tv1 = add(tv0, new Float(2.0));
```

Comment:
This looks cleaner, but wouldn't just using auto tv1 = add(...); or Val* tv1 = add(...); do the same?

The question is a bit deeper than just syntax: why do we want the more specific types? It seems to go against the idea of using inheritance to model the various node types, am I wrong?

Reply:
With `auto tv`, it doesn't allow any calls that require `TensorView*` without type casting. For example, compiling `tv1->split()` would fail because `split` is defined only for `TensorView`.

Reply:
Right, in some cases we care about a more specific subtype. That's not the case for all the current static_casts, right? (my point is that some cleanup is already possible)

I'm curious, what is the main motivation here? Simplify the notation? Enable more (static) checks?

Reply:
It's the former, I believe. See #36

Reply:
Thanks, I missed that. I added a comment directly to the issue.

(I'm not opposed to having convenience overloads, I'm just trying to understand if this is part of a larger plan to move towards more of a ... statically typed IR API?)